### DISCUSSION TITLE
请求HTTP/1.1 404 Not Found returned for "https://api.weixin.qq.com/v3/pay/transactions/jsapi

### DISCUSSION THREAD

**Discussion by xinyi-9** [w7corp/easywechat#2829] [Q&A] [answered]
namespace app\common\service;

use app\common\exception\BaseException;
use EasyWeChat\Kernel\Exceptions\InvalidArgumentException;
use EasyWeChat\MiniApp\Application;

class EasyWechatService
{
    /**
     * 获取小程序实例
     * @throws BaseException
     */
    public static function getApp(): Application
    {
        // 获取当前小程序信息
        $wxConfig = sysConfig('wechat');
        // 验证appid和appsecret是否填写
        if (empty($wxConfig['app_id']) || empty($wxConfig['wxapp_secret']) || empty($wxConfig['mchid'])) {
            throw new BaseException('小程序配置信息未填写完整');
        }
        $config = [
                'app_id' => $wxConfig['app_id'],
                'secret' => $wxConfig['wxapp_secret'],
                'mchid'  => $wxConfig['mchid'],
                'response_type' => 'array',
        ];
        return new Application($config);
    }
}    
$app    = EasyWechatService::getApp();
        $config = $app->getConfig();
        $api    = $app->getClient();
        $data   = [
                "mchid" => $config->get('mchid'), // <---- 请修改为您的商户号
                "out_trade_no" => $orderInfo->order_no,
                'attach' => json_encode(['order_type' => $orderInfo->order_type, 'pay_source' => $orderInfo->platform]),
                "appid" => $config->get('app_id'), // <---- 请修改为服务号的 appid
                "description" => "Image形象店-深圳腾大-QQ公仔",
//                "notify_url" => "https://lct.dtveek.com/api/wxpay", // <---- 请修改为您的回调地址
                "notify_url" => "https://weixin.qq.com/", // <---- 请修改为您的回调地址
                "amount" => [
                        "total" => (int) $orderInfo->amount * 100,
                        "currency" => "CNY"
                ],
                "payer" => [
                        "openid" => $this->user->openid // <---- 请修改为服务号下单用户的 openid
                ]
        ];
        $url = "v3/pay/transactions/jsapi";
        try {
            $response = $api->postJson($url, $data);
        } catch(Exception $e) {
            halt($e->getMessage());
        }
        halt($response->getContent(),$response->getStatusCode()); composer 6.7版本的，按照示例报错，原因在哪儿？

**Comment by TheNorthMemory** [comment]
你的client搞错了，微信支付的base_uri是 `https://api.mch.weixin.qq.com/`

**Reply by lenny168** [reply]
同样问题，使用实例上的`getClient()`， `微信支付的base_uri也是`https://api.weixin.qq.com`，找了一下没找到`https://api.mch.weixin.qq.com/`是否是6.7版本还不支持这个base_uri，只能通过`setClinet()`吗？
```
$response = $app->getClient()->postJson("v3/pay/transactions/jsapi", [
   "mchid" => "1518700000", // <---- 请修改为您的商户号
   "out_trade_no" => "native12177525012012070352333'.rand(1,1000).'",
   "appid" => "wx6222e9f48a0xxxxx", // <---- 请修改为服务号的 appid
   "description" => "Image形象店-深圳腾大-QQ公仔",
   "notify_url" => "https://weixin.qq.com/",
   "amount" => [
        "total" => 1,
        "currency" => "CNY"
    ],
    "payer" => [
        "openid" => "o4GgauInH_RCEdvrrNGrnxxxxxx" // <---- 请修改为服务号下单用户的 openid
    ]
]);

```
\dd($response->toArray(false));

**Reply by TheNorthMemory** [reply, marked as answer]
关键点：

```diff
- use EasyWeChat\MiniApp\Application;
+ use EasyWeChat\Pay\Application;
```
